src/processes/scoring/calc-karma-score.js

```diff
@@ -0,0 +1,79 @@
+const {getDb} = require('../../config/mongodb_conn');
+const {DB_COLLECTION_USER_SCORE} = require('../scoring-constant');
+const UsersFunction = require('../../databases/functions/users');
+
+const KARMA_SCORE_MULTIPLIER = 7.3;
+const KARMA_SCORE_EXPONENT = 0.57;
+
+const calculateP = (p) => {
+  let P = Number(p);
+  P = P === 0 ? 1 : P;
+  return P;
+};
+const combinedUserScore = async (userId, userScoreCol) => {
+  console.log('Start calculating combined user score');
+  const userIds = [userId];
+  const anonymousUserId = await UsersFunction.findAnonymousUserId(userId);
+  if (anonymousUserId) {
+    userIds.push(anonymousUserId);
+  }
+  const users = await userScoreCol.find({_id: {$in: userIds}}).toArray();
+
+  const userExists1 = users[0];
+  const userExists2 = users.length > 1 && users[1];
+  const P1 = userExists1 ? calculateP(users[0].last_posts?.counter) : 1;
+  const P2 = userExists2 ? calculateP(users[1].last_posts?.counter) : 1;
+
+  const u_score1 = typeof users[0].user_score === 'number' ? users[0].user_score : 0;
+  const u_score2 = typeof users[1].user_score === 'number' ? users[1].user_score : 0;
+  // return u_score1;
+  return (u_score1 * P1 + u_score2 * P2) / (P1 + P2);
+};
+
+const percentileUserScore = async (targetScore, userScoreCol) => {
+  console.log('Start calculating percentile user score');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Same as the previous comment about `console.log`. Use a dedicated logging library instead.

```diff
- console.log('Start calculating percentile user score');
+ logger.debug('Start calculating percentile user_score');
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->